WebAPI/Controllers/RolesAndPermissionsController.cs

```diff
@@ -31,11 +84,303 @@ public RolesAndPermissionsController(IRolesAndPermissionsManager rolesAndPermiss
         [ProducesResponseType(403, Type = null)]
         [ProducesResponseType(500, Type = null)]
         [HttpPost("InsertRole")]
-        public IActionResult InsertRole([FromBody] Role role)
+        public IActionResult InsertRole([FromBody] RoleDTO role)
         {
             if (role == null) return BadRequest();
-            var resRole = _rolesAndPermissionsManager.InsertRole(role);
-            return StatusCode(StatusCodes.Status201Created, resRole.Id);
+            return StatusCode(StatusCodes.Status201Created, _rolesAndPermissionsManager.InsertRole(role));
+        }
+
+        /// <summary>
+        ///     removes a role from the DB
+        /// </summary>
+        /// <param name="roleId">Role's Id to be deleted </param>
+        /// <response code="500"></response>
+        /// <response code="403">UnAuthorized Access </response>
+        /// <response code="202">Role Deleted</response>
+        [ProducesResponseType(202, Type = null)]
+        [ProducesResponseType(403, Type = null)]
+        [ProducesResponseType(500, Type = null)]
```

Comment:
So, what is the point of this really? you are just saying that the server may return 500 despite you never actually return it in your code? does the user not know by default the the server may always return 500, 400 and 401?